firebase-perf/src/main/java/com/google/firebase/perf/metrics/FrameMetricsCalculator.java

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.perf.metrics;
+
+import android.util.SparseIntArray;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.app.FrameMetricsAggregator;
+import com.google.firebase.perf.util.Constants;
+
+/**
+ * FrameMetricsCalculator helps calculate total frames, slow frames, and frozen frames from metrics
+ * collected by {@link FrameMetricsAggregator}
+ *
+ * @hide
+ */
+public class FrameMetricsCalculator {
+  public static class FrameMetrics {
```

Comment:
COuld this be namespaced to something with a "Perf prefix"? FrameMetrics is an Android class name as well. [Reference](https://developer.android.com/reference/android/view/FrameMetrics)

Reply:
Done!